test/WaterDrops.ts

```diff
@@ -93,76 +91,83 @@ describe("WaterDrops", function () {
     App = await ethers.getContractFactory("WaterDrops", admin);
 
     waterDrops = await App.deploy(
-        sf.settings.config.hostAddress,
-        sf.settings.config.cfaV1Address
+      sf.settings.config.hostAddress,
+      sf.settings.config.cfaV1Address
     );
 
     await waterDrops.deployed();
 
     // Make Some RICx tokens
-    await ric.mint(
-      admin.address, ethers.utils.parseEther("10000000")
-    );
-    await ric.connect(admin).approve(ricx.address, ethers.utils.parseEther("10000000"));
+    await ric.mint(admin.address, ethers.utils.parseEther("10000000"));
+    await ric
+      .connect(admin)
+      .approve(ricx.address, ethers.utils.parseEther("10000000"));
 
     let ricxUpgradeOperation = ricx.upgrade({
-      amount: ethers.utils.parseEther("10000000")
+      amount: ethers.utils.parseEther("10000000"),
     });
     await ricxUpgradeOperation.exec(admin);
 
     // Transfer RICx to the waterdrops contract
     let transferOperation = ricx.transfer({
       receiver: waterDrops.address,
-      amount: ethers.utils.parseEther("10000000")
+      amount: ethers.utils.parseEther("10000000"),
     });
     await transferOperation.exec(admin);
-
-
   });
 
-  beforeEach(async function() {
-
-  });
+  beforeEach(async function () {});
 
   it("#1.1 - Create a new claimable waterdrop", async function () {
     // As owner, create a new Claim
     let duration = 3600; // one hour
-    let rate = 1000000;  // tokens per second to claim
+    let rate = 1000000; // tokens per second to claim
     let deadline = (await currentBlockTimestamp()) * 7200; // 2 hours
-    await waterDrops.addClaim(ricx.address, rate, duration, deadline, {from: admin.address});
+    await waterDrops.addClaim(ricx.address, rate, duration, deadline, {
+      from: admin.address,
+    });
 
     // verify claim was made and saved correctly
-    let claim = await waterDrops.claims(1, {from: admin.address});
+    let claim = await waterDrops.claims(1, { from: admin.address });
     expect(claim.token).to.equal(ricx.address);
     expect(claim.rate).to.equal(rate);
     expect(claim.duration).to.equal(duration);
     expect(claim.deadline).to.equal(deadline);
 
     // Call the addClaim function and expect the NewClaim event to be emitted
     await expect(
-      waterDrops.addClaim(ricx.address, rate, duration, deadline, {from: admin.address})
-    ).to.emit(
-      waterDrops,
-      "NewClaim"
-    ).withArgs(ricx.address, rate, duration, deadline);
+      waterDrops.addClaim(ricx.address, rate, duration, deadline, {
+        from: admin.address,
+      })
+    )
+      .to.emit(waterDrops, "NewClaim")
+      .withArgs(ricx.address, rate, duration, deadline);
```

Comment:
@hrishibhattu, can you please check for the events in the ConditionalWaterDrop tests?

Reply:
@Greyshaws added this test here, so you can follow her pattern in your tests, I didn't see any tests to verify the events are happening correctly, maybe I missed it

Reply:
I've added the tests for events and also corrected the tests written by @Greyshaws.
There are two tests failing due to error in the contract, which I don't know how to solve.